### PR TITLE
SONARJAVA-1035 RSPEC-2693 Threads should not be started in constructors

### PR DESCRIPTION
**Pull request by merciesa**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/ThreadStartedInConstructorCheck.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  key = "S2693",
+  name = "Threads should not be started in constructors",
+  tags = {"multi-threading", "pitfall"},
+  priority = Priority.CRITICAL)
+@ActivatedByDefault
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.SYNCHRONIZATION_RELIABILITY)
+@SqaleConstantRemediation("30min")
+public class ThreadStartedInConstructorCheck extends SubscriptionBaseVisitor {
+
+  private final static MethodInvocationMatcher THREAD_START = MethodInvocationMatcher.create()
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/ThreadStartedInConstructorCheck.java]: :new: MINORReorder the modifiers to comply with the Java Language Specification.[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 new issue:
* 1 minor

Details:
* [Reorder the modifiers to comply with the Java Language Specification.](https://github.com/SonarSource/sonar-java/blob/3cf4d3aeb5f193c6047d8f3d5a8322d2c96fc97a/java-checks/src/main/java/org/sonar/java/checks/ThreadStartedInConstructorCheck.java#L50)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by benzonico** [comment]
LGTM

**Pull request closed without merging**